src/vehicles/RosVehicle.lua

```diff
@@ -50,86 +50,153 @@ end
+    spec.l_v_x_0 = 0
+    spec.l_v_y_0 = 0
+    spec.l_v_z_0 = 0
+    spec.sec = 0
+
+    -- laser scanner initialization
+
+    -- if there is no custom laser scanner setting for this vehicle, use the default settings to initialize an object of LaserScanner class
+    -- note: a laser scanner is always mounted in the default settings
+    if not mod_config.vehicle[spec.ros_veh_name] then
+        spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle["default_vehicle"])
+    -- if the custom laser scanner is mounted (parameter enabled = true), initialize an object of LaserScanner class
+    elseif mod_config.vehicle[spec.ros_veh_name].laser_scan.enabled then
+        spec.laser_scan_obj = LaserScanner.new(self, mod_config.vehicle[spec.ros_veh_name])
+
+    end
+
+    --  only if the object of LaserScanner class was initialized (the laser scanner is enabled), initialize camera settings
+    if spec.laser_scan_obj then
+        -- store the individual vehicle config file in the specialization to avoid reloading in the loop
+        spec.instance_veh = VehicleCamera:new(self, RosVehicle)
+        spec.xml_path = spec.instance_veh.vehicle.configFileName
+        spec.xmlFile = loadXMLFile("vehicle", spec.xml_path)
+        --  get the cameraRaycast node 2(on top of ) which is 0 index .raycastNode(0)
+        --  get the cameraRaycast node 3 (in the rear) which is 1 index .raycastNode(1)
+        local cameraKey = string.format("vehicle.enterable.cameras.camera(%d).%s", 0, spec.laser_scan_obj.vehicle_table.laser_scan.laser_attachments)
+        XMLUtil.checkDeprecatedXMLElements(spec.xmlFile, spec.xml_path, cameraKey .. "#index", "#node") -- FS17 to FS19
+        local camIndexStr = getXMLString(spec.xmlFile, cameraKey .. "#node")
+        spec.instance_veh.cameraNode =
+            I3DUtil.indexToObject(
+            spec.instance_veh.vehicle.components,
+            camIndexStr,
+            spec.instance_veh.vehicle.i3dMappings
+        )
+        if spec.instance_veh.cameraNode == nil then
+            print("nil camera")
+
+        end
+
+        -- create laser_frame_1 attached to raycastNode (x left, y up, z into the page)
+        -- and apply a transform to the self.laser_frame_1
+        local tran_x, tran_y, tran_z = spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.x, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.y, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.translation.z
+        local rot_x, rot_y, rot_z = spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.x, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.y, spec.laser_scan_obj.vehicle_table.laser_scan.laser_transform.rotation.z
+        local laser_frame_1 = frames.create_attached_node(spec.instance_veh.cameraNode, self:getFullName(), tran_x, tran_y, tran_z, rot_x, rot_y, rot_z)
+        spec.LaserFrameNode = laser_frame_1
```

Comment:
if `cameraNode == nil`, can the `laser_frame_1` be initialised here?

Reply:
yes and no, it still can be initialized at the origin in the FS world but it's not attached/linked to the `cameraNode`. There will be warning  at this [line](https://github.com/tud-cor/FS19_modROS/blob/master/src/utils/frames.lua#L26).

> Warning (script): 'link': Argument 1 has wrong type. Expected: Int. Actual: Nil

Reply:
So should we then guard against that problem occurring?